### PR TITLE
[Hold] fixed spaces collapsing in Messages

### LINKED ISSUE #3859: Spaces are not displayed after sending the message
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in to e.cash and navigate to a conversation
2. Send the following text 
```
0 space
 1 spaces
  2 spaces
   3 spaces
    4 spaces
```
## Expected Result:
Spaces should be displayed in the message after it's sent 
## Actual Result:
No spaces are displayed in the conversation after sending the message 

## Workaround:
N/A

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App
 Mobile Web

**Version Number:** 1.0.75-5

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

image

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 

From @mallenexpensify https://expensify.slack.com/archives/C01GTK53T8Q/p1625162491337400

> Issue: Spaces at the beginning of lines are removed when sending a message

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

https://github.com/Expensify/Expensify.cash/issues/3859#issuecomment-872606868

### Fixed Issues

$ Fixes #3859

### Tests
1. Do `npm install` to get the latest packages used.
2. Open E.cash local server.
3. Go to any conversation and send the following message.
```
0 space
 1 spaces
  2 spaces
   3 spaces
    4 spaces
```

### QA Steps

 1. Log in to e.cash and navigate to a conversation
 2. Send the following text
```
0 space
 1 spaces
  2 spaces
   3 spaces
    4 spaces
```
3. Output should contain the same amount of spacing as added.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Desktop

image

#### Mobile Web

Screen Shot 2021-07-04 at 07 59 52

#### Android

Screenshot 2021-07-04 07:58:25

### CODE REVIEW

File: src/components/RenderHTML/BaseRenderHTML.js [hunk 1]
```diff
@@ -54,10 +54,14 @@ const EXTRA_FONTS = [
  *
  * @param {number} contentWidth - The content width provided to the HTML
  * component.
+ * @param {number} tagName - tagname
```
Comment (jasperhuangg) [inline src/components/RenderHTML/BaseRenderHTML.js:57]: Can we provide a better description for this parameter?

**Review by jasperhuangg — CHANGES_REQUESTED** [review]
Left a comment about some JSDoc stuff. Also, tested it and it doesn't seem to work, am I missing something? This doesn't seem to happen on main. Were there any changes in expensify-common that I'm missing?

https://user-images.githubusercontent.com/31285285/124532430-259aa900-de43-11eb-8bb9-04708b8da818.mp4

**Comment by parasharrajat** [comment]
did you do 'npm install'?

**Comment by jasperhuangg** [comment]
@parasharrajat Ah forgot, make sure you include that in your testing instructions if there were changes to `expensify-common`.

**Comment by parasharrajat** [comment]
@jasperhuangg  Please, let me know if you are facing some issues while testing.

**Comment by jasperhuangg** [comment]
@parasharrajat Still waiting on you to look into [this](https://github.com/Expensify/Expensify.cash/pull/3869/files#r664190613).

**Comment by parasharrajat** [comment]
Sorry @jasperhuangg, I thought you are still facing the issue so I waiting to get a response over it.
Anyways, I have updated it. Thanks.

File: src/components/RenderHTML/BaseRenderHTML.js [hunk 2]
```diff
@@ -54,10 +54,14 @@ const EXTRA_FONTS = [
  *
  * @param {number} contentWidth - The content width provided to the HTML
  * component.
+ * @param {number} tagName - tagname of the Element whose maxWidth is need to be set
```
Comment (jasperhuangg) [inline src/components/RenderHTML/BaseRenderHTML.js:57]:
```suggestion
 * @param {number} tagName - tagname of the Element whose maxWidth needs to be set
```

**Review by jasperhuangg — DISMISSED** [review]
@parasharrajat Looks good! One small grammatical fix.

**Comment by parasharrajat** [comment]
@jasperhuangg Done. Thanks.

**Review by jasperhuangg — APPROVED** [review]
Awesome, thanks for your patience!

**Comment by jasperhuangg** [comment]
Ah crap, didn't see the HOLD on this. I opened up a revert https://github.com/Expensify/Expensify.cash/pull/3969. Sorry for any inconvenience @parasharrajat!

**Comment by kavimuru** [comment]
## Chat - Spaces are not displayed after sending the message
## Expected Result: 
Spaces should be displayed in the message after it's sent
## Actual Result:
No spaces are displayed in the conversation after sending the message
## Actions Performed:
1. Log in to e.cash and navigate to a conversation
2. Send the following text
0 space
 1 spaces
  2 spaces
   3 spaces
    4 spaces
## Platform:
iOS ✔️
mWeb ✔️
Android ✔️
Web ✔️
## Build:
1.0.78-0
## Notes/Images/Video:
Bug5152224_IMG_6279

https://user-images.githubusercontent.com/43996225/125723624-51227edb-b65c-46e0-9284-59c9d7cfcec9.mp4

**Comment by parasharrajat** [comment]
This PR is reverted. So the original issue still exists.

**Comment by isagoico** [comment]
@jasperhuangg We're good to check off this PR from the list if the PR was reverted right? CC @roryabraham

**Pull request merged**